src/libraries/Common/src/System/Security/Cryptography/DSAAndroid.cs

```diff
@@ -0,0 +1,423 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.IO;
+using Internal.Cryptography;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.Security.Cryptography
+{
+#if INTERNAL_ASYMMETRIC_IMPLEMENTATIONS
+    public partial class DSA : AsymmetricAlgorithm
+    {
+        public static new DSA Create()
+        {
+            return new DSAImplementation.DSAAndroid();
+        }
+    }
+
+    internal static partial class DSAImplementation
+    {
+#endif
```

Comment:
@elinor-fung has already changed all the other asymmetric algorithms for Android to use a partial .Create.cs file for the create method, may as well just start this one off doing that and get rid of the \#ifs.